docker/cloudbuild.yaml

```diff
@@ -2,37 +2,58 @@
 # $ gcloud builds submit
 
 steps:
+# Pre-requisite: build the base OS container; 'temp' is referenced in the
+# Dockerfile for subsequent steps but is not pushed. Building this shared base
+# image enables the versioned builders to share base layers.
+- name: 'gcr.io/cloud-builders/docker'
+  args:
+  - 'build'
+  - '--tag=temp'
+  - '--file=Dockerfile-base'
+  - '.'
+  id: 'base'
+
 # Build all supported versions.
 - name: 'gcr.io/cloud-builders/docker'
   args:
   - 'build'
   - '--tag=gcr.io/$PROJECT_ID/docker:19.03.9'
-  - '--file=Dockerfile-19.03.9'
+  - '--file=Dockerfile-versioned'
+  - '--build-arg=DOCKER_VERSION=5:19.03.9~3-0~ubuntu-focal'
+  - '--build-arg=RELEASE_CHANNEL=edge'
   - '.'
   id: '19.03.9'
+  wait_for: ['base']
 
 - name: 'gcr.io/cloud-builders/docker'
   args:
   - 'build'
   - '--tag=gcr.io/$PROJECT_ID/docker:20.10.13'
-  - '--file=Dockerfile-20.10.13'
+  - '--file=Dockerfile-versioned'
+  - '--build-arg=DOCKER_VERSION=5:20.10.14~3-0~ubuntu-focal'
```

Comment:
We're building 20.10.14 then tagging it 20.10.13.

Reply:
Good catch; fixed.